### PR TITLE
Avoid providing completion when "maxProperties" is reached #269

### PR DESCRIPTION
**Pull request by apupier**
PR to show which cases would be nice to be supported

### CODE REVIEW

**Comment by apupier** [comment]
updated PR providing an implementation for maxProperties respect.

please note that I'm not familiar neither with Yaml, Json schema and this code base, so please review carefully.

File: ploparraymax.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- prop1: ded
+- prop2: ded
+- prop3: plop
```
Comment (JPinkney) [inline ploparraymax.yaml:3]: Is thie file needed anywhere or was it just used to test?

File: test/fixtures/testArrayMaxProperties.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+                "type": "string"
+            },"prop2": {
+                "type": "string"
+            },"prop3": {
+                "type": "string"
+            }
+        },
+        "maxProperties": 2
+    }
+  }
```
Comment (JPinkney) [inline test/fixtures/testArrayMaxProperties.json:17]: newline

File: ploparraymax.yaml (hunk 1 above)
Comment (apupier) [reply ploparraymax.yaml:3]: just for test

Comment (apupier) [reply ploparraymax.yaml:3]: I shouldn't have pushed it!!! my bad. Thanks for pointing it

**Pull request merged**